packages/pglite/src/live/index.ts

```diff
@@ -322,104 +416,148 @@ const setup = async (pg: PGliteInterface, _emscriptenOpts: any) => {
       return {
         fields,
         initialChanges: changes!.rows,
+        subscribe,
         unsubscribe,
         refresh,
-      } satisfies LiveChangesReturn<T>
+      } satisfies LiveChanges<T>
     },
 
     async incrementalQuery<T>(
-      query: string,
-      params: any[] | undefined | null,
-      key: string,
-      callback: (results: Results<T>) => void,
+      query: string | LiveIncrementalQueryOptions<T>,
+      params?: any[] | null,
+      key?: string,
+      callback?: (results: Results<T>) => void,
     ) {
+      let signal: AbortSignal | undefined
+      if (typeof query !== 'string') {
+        signal = query.signal
+        params = query.params
+        key = query.key
+        callback = query.callback
+        query = query.query
+      }
+      if (!key) {
+        throw new Error('key is required for incremental queries')
+      }
+      let callbacks: Array<(results: Results<T>) => void> = callback
+        ? [callback]
+        : []
       const rowsMap: Map<any, any> = new Map()
       const afterMap: Map<any, any> = new Map()
       let lastRows: T[] = []
       let firstRun = true
 
-      const { fields, unsubscribe, refresh } = await namespaceObj.changes<T>(
-        query,
-        params,
-        key,
-        (changes) => {
-          // Process the changes
-          for (const change of changes) {
-            const {
-              __op__: op,
-              __changed_columns__: changedColumns,
-              ...obj
-            } = change as typeof change & { [key: string]: any }
-            switch (op) {
-              case 'RESET':
-                rowsMap.clear()
-                afterMap.clear()
-                break
-              case 'INSERT':
-                rowsMap.set(obj[key], obj)
-                afterMap.set(obj.__after__, obj[key])
-                break
-              case 'DELETE': {
-                const oldObj = rowsMap.get(obj[key])
-                rowsMap.delete(obj[key])
-                afterMap.delete(oldObj.__after__)
-                break
-              }
-              case 'UPDATE': {
-                const newObj = { ...(rowsMap.get(obj[key]) ?? {}) }
-                for (const columnName of changedColumns) {
-                  newObj[columnName] = obj[columnName]
-                  if (columnName === '__after__') {
-                    afterMap.set(obj.__after__, obj[key])
-                  }
+      const runCallbacks = (results: Results<T>) => {
```

Comment:
I feel like this `runCallbacks` function along with the signal abort and unsubscribe logic should be extracted to shared helper functions (like `runCallbacks(results: Results<T>, callbacks: (Results<T>) => void)`)